scanner/src/main/kotlin/Scanner.kt

```diff
@@ -207,6 +207,40 @@ class Scanner(
             }
         }
 
+        val filteredScanResults = filterScanResultsByVcsPaths(controller.getAllScanResults(), vcsPathsForProvenances)
```

Comment:
Ok, a few random starting thoughts / comments:
- This code gets complex and deserves a few more code comments that explain the algorithm / intention.
- For the default case ideally no unnecessary copy should be done (like is done now in line 217). To archive that, I'd leave `val filteredScanResults` as-is in original line 195, and instead introduce a new variable like

        val scanResults = if (scannerConfig.includeUnlicensed) {
            // Do the work.
        } else {
            filteredScanResults
        }

        // ....

        return ScannerRun.EMPTY.copy(
            config = scannerConfig,
            provenances = provenances,
            scanResults = scanResults,
            files = files,
            scanners = scanners
        )